x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/action/service/RestClearServiceAccountTokenStoreCacheAction.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.security.rest.action.service;
+
+import org.elasticsearch.client.node.NodeClient;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.license.XPackLicenseState;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.action.RestActions;
+import org.elasticsearch.xpack.core.security.action.ClearSecurityCacheAction;
+import org.elasticsearch.xpack.core.security.action.ClearSecurityCacheRequest;
+import org.elasticsearch.xpack.security.rest.action.SecurityBaseRestHandler;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.elasticsearch.rest.RestRequest.Method.POST;
+
+public class RestClearServiceAccountTokenStoreCacheAction extends SecurityBaseRestHandler {
+
+    public RestClearServiceAccountTokenStoreCacheAction(Settings settings, XPackLicenseState licenseState) {
+        super(settings, licenseState);
+    }
+
+    @Override
+    public List<Route> routes() {
+        return List.of(new Route(POST, "/_security/service/{namespace}/{service}/credential/token/{name}/_clear_cache"));
+    }
+
+    @Override
+    public String getName() {
+        return "xpack_security_clear_service_account_token_store_cache";
+    }
+
+    @Override
+    protected RestChannelConsumer innerPrepareRequest(RestRequest request, NodeClient client) throws IOException {
+        final String namespace = request.param("namespace");
+        final String service = request.param("service");
+        String[] tokenNames = request.paramAsStringArrayOrEmptyIfAll("name");
+
+        ClearSecurityCacheRequest req = new ClearSecurityCacheRequest().cacheName("service");
+        if (tokenNames.length == 0) {
+            // This is the wildcard case for tokenNames
+            req.keys(namespace + "/" + service + "/");
+        } else {
+            req.keys(Arrays.stream(tokenNames).map(name -> namespace + "/" + service + "/" + name).toArray(String[]::new));
+        }
```

Comment:
This cache clearing URL is different from all the other ones in that the cache key is comprised of three parts, i.e. `{namespace}/{service}/{name}`. This makes it challenging to conform 100% to existing behaviours. For an exmple, it is not possible to use a single `*` for clearing everything, it would be more like `/_security/service/*/*/credential/token/*/_clear_cache`. This is doable but I think it is not really needed for now as it adds complexity for the key comparison logic, e.g. should `namespace` also support comma separated list? 

So for now, I am keep this simple by taking the `namespace` and `service` fields literally and only allow `*` and comma separated list for the token `name` field. Since we only have a single service account now, this approach can still clear all caches with `/_security/service/elastic/fleet-server/credential/token/*/_clear_cache`. We should revisit this once we need to add another service account.

Reply:
Cool. I'll leave it as is then. Always clearing all tokens of a service account can provide some simplication, but not a lot. Since I already have things in place, I'd just keep them. Plus even with the clearing applicaiton privileges cache API, the URL format is something like `.../{application}/_clear_cache` which maps better to `.../{token_name}/_clear_cache`.